tools/orca/src/state.rs

```diff
@@ -112,3 +117,14 @@ pub struct Person {
     pub credential: Credential,
     pub model: Model,
 }
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Group {
```

Comment:
Okay, so. 

I think you should stop and think about this some more because you're on the right track, but this current approach will hit limits pretty quick.

Something we'll need is to have persons with *multiple* roles, and those roles gain permissions from groups. In your current approach every person only gets one actor role but why is that the case?

We create groups, then assign a role to the group, then assign persons to those groups.

This way persons can have M of N roles, and then within the person it can see "I'm in these roles, so I can make these choices or actions". 

So shouldn't this model

Reply:
Yeah in hindsight it was a quite naive attempt :p. What I'd like to better understand before diving deeper is the mechanism that will determine how persons behave. The best idea I've had so far relies on kani to determine who is entitled to do what,  while on the orca side persons don't know (at least at the start) what they can do. Here's my plan: at first we basically let everyone attempt to do everything, and when they get a 403 they don't try that specific action again. After a little while we "reset" their memory of allowed actions and we repeat the process.  We proceed in this fashion until the end of the test. 
The rationale behind this approach is that during the test itself we could be interested in changing some roles, and I couldn't come up with a better way to dynamically fetch what one can do or cannot do. 
What do you think? Should I start coding something that more or less works like this?

Reply:
Rather than adding in the randomness, why not just make some persons that behave in a static manner first?

Reply:
Sure!